src/test/java/IvanFFavoriteTests.java

```diff
@@ -0,0 +1,109 @@
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.annotations.TestInstance;
+import java.time.Duration;
+
+
+public class IvanFFavoriteTests extends BaseTestIvan{
+    @Test
+    public void checkFavorite() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        System.out.println("Number of favorites         "+(homePageIvan.getNumberfavourite()));
+
+    }
+
+
+    @Test
+    public void checkQueue() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        homePageIvan.playSongFromfav();
+        homePageIvan.currentQueue();
+    }
+    @Test
+    public void deleteSongFromFav() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        homePageIvan.deleteSongFromfav();
+
+    }
+    @Test
+    public void addSongToFavHeart() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        int initialNumberOfFav = homePageIvan.getNumberfavourite();
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        int updatedNumberOfFav = homePageIvan.getNumberfavourite();
+        System.out.println("Number of favorites before: "+initialNumberOfFav+" New Number Of Favorites:  "+updatedNumberOfFav);
+
+    }
+    @Test
+    public void downloadAllSongFromFav() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        homePageIvan.favorite();
+        homePageIvan.downloadAllSongFromFav();
+
+    }
+    @Test
+    public void downloadOneSongFromFav() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        homePageIvan.favorite();
+        Thread.sleep(2000);
+        homePageIvan.contextDownload();
+        Thread.sleep(2000);
+        Assert.assertTrue(homePageIvan.isSongDownloaded("HoliznaCC0 - Waiting On A Train.mp3"));
```

Comment:
**"Magic" numbers and strings are constants that have meaning beyond their immediate, literal value.**
Do not use literal values, either numeric or strings, in your code other than to define symbolic constants. Use the following pattern to define variable:

For example

```
    String nameSong = "HoliznaCC0 - Waiting On A Train.mp3";
    Assert.assertTrue(homePageIvan.isSongDownloaded(nameSong));
```